Makefile

```diff
@@ -15,42 +15,41 @@
 GOOS ?= $(shell go env GOOS)
 GOARCH ?= $(shell go env GOARCH)
 BUILD_DIR ?= ./out
-GOPATH := $(shell pwd)/.gopath
+GOPATH ?= $(shell pwd)/.gopath
 REPOPATH ?= k8s.io/minikube
 export GO15VENDOREXPERIMENT=1
 
 clean:
 	rm -rf $(GOPATH)
 	rm -rf $(BUILD_DIR)
 
-gopath:
+.gopath:
 	mkdir -p $(shell dirname $(GOPATH)/src/$(REPOPATH))
 	ln -s -f $(shell pwd) $(GOPATH)/src/$(REPOPATH)
 
-.PHONY: minikube
-minikube: minikube-$(GOOS)-$(GOARCH)
+out/minikube: out/minikube-$(GOOS)-$(GOARCH)
 	cp $(BUILD_DIR)/minikube-$(GOOS)-$(GOARCH) $(BUILD_DIR)/minikube
 
-.PHONY: localkube
-localkube: localkube-$(GOOS)-$(GOARCH)
-	cp $(BUILD_DIR)/localkube-$(GOOS)-$(GOARCH) $(BUILD_DIR)/localkube
+out/localkube: .gopath
+ifeq ($(GOOS),linux)
+	CGO_ENABLED=1 go build -v -ldflags="-s" -o $(BUILD_DIR)/localkube ./cmd/localkube
```

Comment:
The `-v` here makes it very verbose all the time. 
Maybe add a `VERBOSE` flag to the makefile so we may control it?
I may do this later as well.